ui/src/lib/gql/queries/workspace.ts

```diff
@@ -0,0 +1,114 @@
+import { useMutation, useQuery, useQueryClient } from "@tanstack/react-query";
+
+import { CreateWorkspaceMutation, useGraphQLContext } from "@flow/lib/gql";
+
+import { graphql } from "../__gen__";
+
+graphql(`
+  mutation CreateWorkspace($input: CreateWorkspaceInput!) {
+    createWorkspace(input: $input) {
+      workspace {
+        id
+      }
+    }
+  }
+`);
+
+// TODO: Should this a fragment in GET_ME?
+graphql(`
+  query GetWorkspaces {
+    me {
+      workspaces {
+        id
+        name
+        personal
+      }
+    }
+  }
+`);
+
+graphql(`
+  mutation UpdateWorkspace($input: UpdateWorkspaceInput!) {
+    updateWorkspace(input: $input) {
+      workspace {
+        id
+        name
+      }
+    }
+  }
+`);
+
+graphql(`
+  mutation DeleteWorkspace($input: DeleteWorkspaceInput!) {
+    deleteWorkspace(input: $input) {
+      workspaceId
+    }
+  }
+`);
+
+export enum WorkspaceQueryKeys {
+  GetWorkspace = "getWorkspace",
+}
+
+type mutationInput = {
+  onSuccess?: () => void;
+  onError?: () => void;
+};
+
+export const useCreateWorkspaceMutation = ({
```

Comment:
Hmm something about all the mutations seem not good to use, and/or limited in their usefulness.
I think passing onSuccess and onError isn't needed. We want these custom hooks to reduce duplicate code, so if we can take in the mutation's input, handle the mutation, and send the results (data, status, etc) like we are doing with the useGetWorkspaceQuery, I think the api will be more consistent and easier to use, etc.
This is the way I wrote it in reearth-viz too, so I am a little biased. If you see your approach to mutations better or see a better solution please let me know and we can discuss.

Reply:
Hmm...we need a way to know if the mutation was successful or not, I'll look into something which is less code

Reply:
At least with apollo, the mutation would return `data` and then other info like whether it was successful or not, etc, which is why we return in the custom hook usually `{ data, ...rest }`

Reply:
This is done. Using `useEffect` now 
```
  useEffect(() => {
    if (createWorkspace.isIdle || createWorkspace.isPending) return;

    if (createWorkspace.isError) {
      setShowError(true);
      setButtonDisabled(false);
    } else if (createWorkspace.isSuccess) {
      const workspaceId = createWorkspace.data?.createWorkspace?.workspace.id;
      setButtonDisabled(false);
      setShowError(false);
      setDialogType(undefined);
      navigate({ to: `/workspace/${workspaceId}` });
    }
  }, [createWorkspace, setShowError, setButtonDisabled, setDialogType, navigate]);

  const handleClick = () => {
    if (!name) return;
    createWorkspace.mutate({ input: { name } });
    setButtonDisabled(true);
  };
```